packages/common/src/utils/analytics.ts

```diff
@@ -5,10 +5,12 @@ import type { PlaylistItem, Source } from '../../types/playlist';
 
 export const attachAnalyticsParams = (item: PlaylistItem) => {
```

Comment:
Not related to your change, but this function mutates the given item, should we create a new object instead?

```ts
export const attachAnalyticsParams = (item: PlaylistItem) => {
  const updatedSources = item.sources.map(source => {
    // logic
    return { ...source, file: url.toString() };
  });

  return { ...item, sources: updatedSources };
};
```